HarmonyOS/Day2 Assingment/Alpesh/assignment_app/entry/src/main/java/com/example/assignment_app/slice/Signup.java

```diff
@@ -0,0 +1,112 @@
+package com.example.assignment_app.slice;
+
+import com.example.assignment_app.ResourceTable;
+import ohos.aafwk.ability.AbilitySlice;
+import ohos.aafwk.content.Intent;
+import ohos.agp.components.Component;
+import ohos.agp.components.RadioButton;
+import ohos.agp.components.Text;
+import ohos.agp.components.TextField;
+
+public class Signup extends AbilitySlice {
+    @Override
+    protected void onStart(Intent intent) {
+        super.onStart(intent);
+        super.setUIContent(ResourceTable.Layout_signup);
+        TextField firstname = (TextField) findComponentById(ResourceTable.Id_firstname);
+        TextField lastname = (TextField) findComponentById(ResourceTable.Id_lastname);
+        TextField email = (TextField) findComponentById(ResourceTable.Id_email);
+        TextField password = (TextField) findComponentById(ResourceTable.Id_password);
+        TextField mobile = (TextField) findComponentById(ResourceTable.Id_mobile);
+        RadioButton male = (RadioButton) findComponentById(ResourceTable.Id_gender_male);
+        RadioButton female = (RadioButton) findComponentById(ResourceTable.Id_gender_female);
+
+        firstname.setFocusChangedListener((component, isFocused) -> {
+            String name = firstname.getText();
+            if (!isFocused) {
+                if (!name.matches("^[a-zA-Z]*$")) {
+                    //firstname.setText("");
+                    Text er1 = (Text) findComponentById(ResourceTable.Id_er1);
+                    er1.setVisibility(Component.VISIBLE);
```

Comment:
"er1" must have a null check as setFocusChangedListener is being used.!! App crash possible scenario : If user clicks on the textfield with setFocusChangedListener and then clicks back button. Make sure all the error fields in this class are having the null checks as they are inflated inside the focusChangedListener

Reply:
Checked